acceptance/tests/resource/service/smf_basic_tests.rb

```diff
@@ -1,61 +1,92 @@
-  clean agent, :service => 'tstapp'
-  manifest, method = setup agent, :service => 'tstapp'
-
-  step "SMF: ensure it is created with a manifest"
-  apply_manifest_on(agent, 'service {tstapp : ensure=>running, manifest=>"%s"}' % manifest) do
-    assert_match( /defined 'ensure' as 'running'/, result.stdout, "err: #{agent}")
-  end
-
-  step "SMF: verify with svcs that the service is online"
-  on agent, "svcs -l application/tstapp" do
-    assert_match( /state\s+online/, result.stdout, "err: #{agent}")
+test_name "SMF: basic tests" do
+  confine :to, :platform => 'solaris'
+  
+  tag 'audit:medium',
+      'audit:refactor',  # Use block style `test_run`
+      'audit:acceptance' # Could be done at the integration (or unit) layer though
+                         # actual changing of resources could irreparably damage a
+                         # host running this, or require special permissions.
+  
+  require 'puppet/acceptance/solaris_util'
+  require 'puppet/acceptance/service_utils'
+  extend Puppet::Acceptance::SMFUtils
+  extend Puppet::Acceptance::ServiceUtils
+  
+  def assert_svcs_info_matches_on(agent, service, info_hash)
+    info_hash.merge({ 'next_state' => 'none' })
+    on(agent, "svcs -l #{service}") do |result|
+      info_hash.each do |key, value|
+        escaped_key, escaped_value = Regexp.escape(key), Regexp.escape(value)
+  
+        assert_match(
+          /^#{escaped_key}.*#{escaped_value}$/,
+          result.stdout,
+          "`svcs -l #{service}` does not indicate that #{key} = #{value} on #{agent}"
+        )
+      end
+    end
   end
-
-  step "SMF: ensure it is idempotent - ie not created again"
-  apply_manifest_on(agent, 'service {tstapp : ensure=>running, manifest=>"%s"}' % manifest, :catch_changes => true)
-
-  step "SMF: ensure you can query the service with the ral"
-  on(agent, puppet("resource service tstapp")) do
-    assert_match( /ensure => 'running'/, result.stdout, "err: #{agent}")
+  
+  teardown do
+    agents.each do |agent|
+      clean(agent, :service => 'tstapp')
+    end
   end
-
-  step "SMF: ensure non-existent services return :absent"
-  on(agent, puppet("resource service bogus")) do
-    assert_match( /ensure => 'absent'/, result.stdout, "err: #{agent}")
-  end
-
-  step "SMF: ensure you can stop the service"
-  apply_manifest_on(agent, 'service {tstapp : ensure=>stopped}') do
-    assert_match( /changed 'running'.* to 'stopped'/, result.stdout, "err: #{agent}")
-  end
-
-  step "SMF: ensure stopping a non-existent service doesn't throw an error"
-  apply_manifest_on(agent, 'service {bogus : ensure=>stopped}', :acceptable_exit_codes => [0,1,2,4,6]) do
-    assert_equal(exit_code, 0, "'puppet resource service' should have an exit code of 0")
-  end
-
-  step "SMF: verify with svcs that the service is not online"
-  on agent, "svcs -l application/tstapp", :acceptable_exit_codes => [0,1] do
-    assert_no_match( /state\s+online/, result.stdout, "err: #{agent}")
+  
+  agents.each do |agent|
```

Comment:
Do these acceptance tests seem enough? I could move over some of the unit tests here. They did help me catch an important bug with the maint state (where it turns out you had to clear it to disable the service too, as well as enabling it).